src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -532,6 +533,10 @@ private function createEncoder($config, ContainerBuilder $container)
             return new Reference($config['id']);
         }
 
+        if ('auto' === $config['algorithm']) {
+            $config['algorithm'] = SodiumPasswordEncoder::isSupported() ? 'sodium' : 'native';
```

Comment:
is the sodium encoder superior to our native encoder? Is that why it's preferred?

Reply:
yes, in two aspects: it is faster for the same cost parameters, and it can be more up to date than the native one, as an extension can move faster to new algos.

Reply:
being faster for the same cost may not actually be superior. The whole point of a password hashing algorithm is that it should not be too fast.

Reply:
> being faster for the same cost may not actually be superior. The whole point of a password hashing algorithm is that it should not be too fast.

this answer is wrong ;)
what matters is that *attackers* must spend significant resources to compute password hashes when they leaked. Being faster for the same cost is just improving UX and reducing greenhouse emissions. So yes, it matters a lot.